commonTools/build_stats/BuildStatsWrappers.cmake

```diff
@@ -48,6 +52,63 @@ function(generate_build_stats_wrappers)
 endfunction()
 
 
+# Generate the build stats compiler wrapper for a given CMake variable.
+#
+# the only difference between this and lang, is setting the proper
+# Cmake variable name, e.g., CMAKE_LANG_COMPILER  vs CMAKE_OP
+# we can resolve this by taking in name of the variable to set
+#
+# The intent of this function is pass in arbitrary cmake variables that
+# map to commands and generate suitable wrappers.
+#
+# Supported functions are C, CXX, Fortran, AR, LD, and RANLIB
+#
+#  TODO: wrap MPIEXEC!
```

Comment:
Yes - it would entail a different file most likely, maybe something like `testing statistics`.  It's something I'll probably poke down the road.  The basic concept of `wrapping` will work, but you'd shove a timing mechanism before the binary (I already do something similar for ATDM EMPIRE so that their process stdout/stderr get mapped to `rank specific files`). 

This would be something for months down the road - but if would help know `how many or which tests` can be run concurrently.

I'd say keep this kinda stuff in mind ... if you think of features/things you'd like let me know. (even if just day dreaming)